spec/os_mapping_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'spec_helper'
+
+LKP_SRC_PATH = "#{LKP_SRC}/distro/adaptation".freeze
+
+def find_duplicated_lines(default_subdir, os_specific_subdir)
+  default_mapping_lines = read_mappings(default_subdir)
+  os_specific_mapping_lines = read_mappings(os_specific_subdir)
+
+  default_mapping_lines & os_specific_mapping_lines
+end
+
+def read_mappings(file_path)
+  File.readlines(file_path).map(&:strip).reject(&:empty?)
+end
+
+def generate_mappings(lkp_src)
+  mappings = Hash.new { |h, k| h[k] = [] }
+
+  # Iterate through each subdirectory in the lkp_src directory
+  Dir.entries(lkp_src).each do |subdir|
+    next if subdir == '.' || subdir == '..' || subdir == 'README.md'
+
+    subdir_path = File.join(lkp_src, subdir)
+    next unless File.directory?(subdir_path)
+
+    # Get a list of files in the subdirectory
+    files = Dir.entries(subdir_path).select do |f|
+      file_path = File.join(subdir_path, f)
+      File.file?(file_path) && !File.symlink?(file_path)
+    end
+
+    # Populate the mappings hash with files from the subdirectory
+    files.each do |file|
+      base_name = subdir
+      if file == base_name
+        mappings[base_name].unshift(File.join(subdir_path, file))
+      else
+        mappings[base_name] << File.join(subdir_path, file)
+      end
+    end
+  end
+  
+  mappings
+end
+
+describe 'package mapping uniqueness' do
+  grouped_files = generate_mappings(LKP_SRC_PATH)
+
+  grouped_files.each_value do |files|
+    default_file = files.first
+    os_specific_files = files[1..]
+
+    os_specific_files.each do |os_specific_file|
+      it "does not have duplicated lines between #{default_file} and #{os_specific_file}" do
+        duplicated_lines = find_duplicated_lines(default_file, os_specific_file)
+        expect(duplicated_lines).to be_empty, "Duplicated lines found between files:\nDefault: #{default_file}\nOS-specific: #{os_specific_file}\nDuplicated lines: #{duplicated_lines.join("\n")}"
+      end
+    end
+
+    os_specific_files.combination(2).each do |os_specific_file_1, os_specific_file_2|
```

Comment:
thanks, need do one more change that the comparison is not between 2 files, because it's ok that there's same mapping in debian/10 and debian/11 (but not in 12). It is not ok if there's mapping existing in all debian os-specific files (10, 11, 12), because in this case, the map need be moved to default file.